### PR TITLE
Suggest a website form now implemented

### PR DESCRIPTION
**Pull request by vinyll**

### CODE REVIEW

**Comment by n1k0** [comment]
The `utils` module is missing ;)

File: uxperiment/templates/websites/list.html [hunk 1]
```diff
@@ -8,7 +8,10 @@
     <li>
         {{ website.title }}
         <a href="#" taget="_blank">{{ website.title }}</a>
-        <img src="/media/{{ website.picture }}" alt="{{ website.title }} ({{ website.url }})" />
+        <img src="/media/{{ website.picture }}" \
```
Comment (davidbgk) [inline uxperiment/templates/websites/list.html:11]: /media/ should not be hard-coded in templates, see https://docs.djangoproject.com/en/dev/howto/static-files/#referring-to-static-files-in-templates

File: websites/views.py [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+
 from websites.models import get_url_informations
 from websites.forms import SuggestForm
+from models import Website
 from uxperiment.utils import send_message
 
-from models import Website
 
 
 def list(request):
```
Comment (n1k0) [inline websites/views.py:17]: I know it's not you but naming our own stuff with language native keywords is never a god idea, here `list` will override the `list` type in current scope.

Poke @tricky21

File: websites/views.py [hunk 3]
```diff
@@ -20,21 +21,22 @@ def list(request):
...
     Display and proceed suggest a website form submission
     """
     if request.method == 'POST':
-        form = SuggestForm(request.POST)
+        form = SuggestForm(request.POST, request.FILES)
         if form.is_valid():
             website = form.save()
-            data = {'website': website.url, 'username': request.user.username,
-                    'sender': request.user.email}
+            data = dict(
```
Comment (n1k0) [inline websites/views.py:33]: `data` is a bit vague; here something like `mail_context` would be more meaningful I guess.

File: uxperiment/templates/websites/list.html (hunk 1 above)
Comment (n1k0) [inline uxperiment/templates/websites/list.html:11]: Also, why is the backslash at the end of the line?

File: uxperiment/templates/websites/list.html [hunk 4]
```diff
@@ -8,7 +8,10 @@
...
+            alt="{{ website.title }} ({{ website.url }})"
+            width="240" height="160"
```
Comment (n1k0) [inline uxperiment/templates/websites/list.html:13]: Shouldn't this be handled through css styles?

File: websites/tests.py [hunk 5]
```diff
@@ -1,16 +1,55 @@
...
+import os
 
-class SimpleTest(TestCase):
-    def test_basic_addition(self):
-        """
-        Tests that 1 + 1 always equals 2.
-        """
-        self.assertEqual(1 + 1, 2)
+
+class WebsiteFixtures(object):
```
Comment (n1k0) [inline websites/tests.py:12]: As a side note, for model mocks & dynamic fixtures generation, [FactoryBoy](https://github.com/dnerdy/factory_boy) is a great lib for that purpose.

File: websites/tests.py [hunk 6]
```diff
@@ -1,16 +1,55 @@
...
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(self.website.url, response.content)
+        self.assertIn(self.website.title, response.content)
+
+class SubmitWebsiteTest(TestCase, WebsiteFixtures):
+    def setUp(self):
+        self.client = Client()
+    def test_suggest_website(self):
+        response = self.client.post('/proposer-un-site/', self.get_config())
+        # FIXME : client.post() has no way to post a file
```
Comment (n1k0) [inline websites/tests.py:47]: Not sure we should keep TODOs, FIXMEs and commented code. As we're in an in-progress PR, not a real problem for now though :)

Comment (n1k0) [inline websites/tests.py:47]: Also, to post a file using the django client: https://docs.djangoproject.com/en/dev/topics/testing/#django.test.client.Client.post

``` python
>>> c = Client()
>>> with open('wishlist.doc') as fp:
...     c.post('/customers/wishes/', {'name': 'fred', 'attachment': fp})
```

**Comment by n1k0** [comment]
Ça va conflicter sévèrement sur master… si tu veux on voit ça cet AM ensemble :)

Edit :  je m'en occupe, c'ets moi qui ait mis le dawa ;)

File: websites/views.py (hunk 2 above)
Comment (MathieuTricoire) [inline websites/views.py:17]: I will fix it

**Pull request merged**